http://mail.python.org/pipermail/python-list/2007-July/451163.html
(OpenBSD 4.0 appears to need it as well)

I see. The macros _BSD_SOURCE didn't actually get defined.
Please try the revised patch below.

Regards,
Martin

Index: configure
===================================================================
--- configure   (Revision 56599)
+++ configure   (Arbeitskopie)
@@ -1388,6 +1388,14 @@
 _ACEOF


+# OpenBSD undoes our definition of __BSD_VISIBLE if _XOPEN_SOURCE is
+# also defined. This can be overridden by defining _BSD_SOURCE
+
+cat >>confdefs.h <<\_ACEOF
+#define _BSD_SOURCE 1
+_ACEOF
+
+
 # The later defininition of _XOPEN_SOURCE and _POSIX_C_SOURCE disables
 # u_int on Irix 5.3. Defining _BSD_TYPES brings it back.

Index: pyconfig.h.in
===================================================================
--- pyconfig.h.in       (Revision 56599)
+++ pyconfig.h.in       (Arbeitskopie)
@@ -909,6 +909,9 @@
 # undef _ALL_SOURCE
 #endif

+/* Define on OpenBSD to activate all library features */
+#undef _BSD_SOURCE
+
 /* Define on Irix to enable u_int */
 #undef _BSD_TYPES
